lib/streamlit/errors.py

```diff
@@ -133,3 +136,249 @@ def __init__(self, module_name, *args):
+            page=page,
+        )
+
+
+# policies
+class StreamlitFragmentWidgetsNotAllowedOutsideError(LocalizableStreamlitException):
+    """Exception raised when the fragment attempts to write to an element outside of its container."""
+
+    def __init__(self):
+        super().__init__("Fragments cannot write widgets to outside containers.")
+
+
+class StreamlitInvalidFormCallbackError(LocalizableStreamlitException):
+    """Exception raised a `on_change` callback is set on any element in a form except for the `st.form_submit_button`."""
+
+    def __init__(self):
+        super().__init__(
+            "Within a form, callbacks can only be defined on st.form_submit_button. "
+            "Defining callbacks on other widgets inside a form is not allowed."
+        )
+
+
+class StreamlitWidgetValueAssignmentNotAllowedError(LocalizableStreamlitException):
+    """Exception raised when trying to set values for a widget where writes are not allowed."""
+
+    def __init__(self, key: str):
+        super().__init__(
+            "Values for the widget with `key` '{key}' cannot be set using `st.session_state`.",
+            key=key,
+        )
+
+
+# st.multiselect
+class StreamlitSelectionCountExceedsMaxError(LocalizableStreamlitException):
+    """Exception raised when there are more default selections specified than the max allowable selections."""
+
+    def __init__(self, current_selections_count: int, max_selections_count: int):
+        super().__init__(
+            "Multiselect has {current_selections_count} {current_selections_noun} "
+            "selected but `max_selections` is set to {max_selections_count}. "
+            "This happened because you either gave too many options to `default` "
+            "or you manipulated the widget's state through `st.session_state`. "
+            "Note that the latter can happen before the line indicated in the traceback. "
+            "Please select at most {max_selections_count} {options_noun}.",
+            current_selections_count=current_selections_count,
+            current_selections_noun="option"
+            if current_selections_count == 1
+            else "options",
+            max_selections_count=max_selections_count,
+            options_noun="option" if max_selections_count == 1 else "options",
+        )
+
+
+# st.columns
+class StreamlitInvalidColumnSpecError(LocalizableStreamlitException):
+    """Exception raised when no weights are specified, or a negative weight is specified."""
+
+    def __init__(self):
+        super().__init__(
+            "The spec argument to `st.columns` must be either a "
+            "positive integer (number of columns) or a list of positive numbers (width ratios of the columns). "
+            "See [documentation](https://docs.streamlit.io/develop/api-reference/layout/st.columns) "
+            "for more information."
+        )
+
+
+# st.set_page_config
+class StreamlitSetPageConfigMustBeFirstCommandError(LocalizableStreamlitException):
```

Comment:
We've had several interested parties at just removing this restriction, especially since it would simplify some things with MPA v2. I think Thiago mentioned it might have been artificially introduced over concern that initial load might look bad if it jumps. I think it'd be nice to remove the restriction and just recommend to users to call the command as early as possible on the page.